src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -1878,6 +1878,20 @@ void VM_Version::check_virtualizations() {
   }
 }
 
+// avx3_threshold() sets the threshold at which 64-byte instructions are used
+// for implementing the array copy and clear operations.
+// The Intel platforms that supports the serialize instruction
+// has improved implementation of 64-byte load/stores and so the default
+// threshold is set to 0 for these platforms.
+int VM_Version::avx3_threshold() {
+  if (FLAG_IS_DEFAULT(AVX3Threshold)) {
+    return ((is_intel_family_core() &&
+             supports_serialize()) ? 0: AVX3Threshold);
+  } else {
+    return AVX3Threshold;
+  }
+}
```

Comment:
I am somewhat concerned about the overhead of evaluating this each time it is used. I realize these will only be startup costs while generating the stubs, not part of the stubs themselves, but it still may be a startup impact. Can you run a startup benchmark to see if there is any problem?

I was also thinking the more direct formulation would just be:
```return (is_intel_family_core() && supports_serialize() && FLAG_IS_DEFAULT(AVX3Threshold)) ? 0 : AVX3Threshold;```

Reply:
Hi  @sviswa7 agree with @dholmes-ora , instead of calling multiple times in a stub can we not call it only once per stub? Since stubs are assembled once and not relocated hence it should be okay to call this method only once for stubs which are going to benefit form this change.